packages/core/src/components/Badge.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react";
+import { View, Text, StyleSheet, TextStyle } from "react-native";
+
+import { withTheme } from "../theming";
+import theme from "../styles/DefaultTheme";
+
+const CONSTANTS = {
+  baseSize: 20,
+};
+
+type BaseProps = {
+  title?: string;
+  style?: TextStyle;
+  size?: number;
+  theme: typeof theme;
+};
+
+function Badge({
+  title,
+  style,
+  size = CONSTANTS.baseSize,
+  ...rest
+}: BaseProps): JSX.Element {
+  const {
+    color,
+    fontFamily,
+    fontWeight,
+    fontSize,
+    lineHeight,
+    letterSpacing,
+    textTransform,
+    textAlign,
+    textDecorationLine,
+    textDecorationColor,
+    textDecorationStyle,
+    ...viewStyles
+  } = StyleSheet.flatten(style || ({} as TextStyle));
```

Comment:
We have an `extractStyles` utility that splits off the text styles and view styles. You don't need to do it manually here.